onnxruntime/core/providers/dml/DmlExecutionProvider/src/Operators/DmlOperatorQAttention.cpp

```diff
@@ -205,52 +189,23 @@ class DmlOperatorQAttention : public DmlOperator
-        inputDequantizeOperatorDesc.InputTensor = &inputDescs[InputIndex::inputIndex];
-        inputDequantizeOperatorDesc.ScaleTensor = &inputDescs[InputIndex::inputScaleIndex];
-        inputDequantizeOperatorDesc.ZeroPointTensor = &inputDescs[InputIndex::inputZeroPointIndex];
-        inputDequantizeOperatorDesc.OutputTensor = &namedIntermediateInputTensorDesc;
-
-        const DML_OPERATOR_DESC inputDequantizeOpDesc{DML_OPERATOR_ELEMENT_WISE_DEQUANTIZE_LINEAR, &inputDequantizeOperatorDesc};
-
-        DML_ELEMENT_WISE_DEQUANTIZE_LINEAR_OPERATOR_DESC weightDequantizeOperatorDesc = {};
-        weightDequantizeOperatorDesc.InputTensor = &inputDescs[InputIndex::weightsIndex];
-        weightDequantizeOperatorDesc.ScaleTensor = &inputDescs[InputIndex::weightScaleIndex];
-        weightDequantizeOperatorDesc.ZeroPointTensor = &inputDescs[InputIndex::weightZeroPointIndex];
-        weightDequantizeOperatorDesc.OutputTensor = &namedIntermediateWeightTensorDesc;
-
-        const DML_OPERATOR_DESC weightDequantizeOpDesc{DML_OPERATOR_ELEMENT_WISE_DEQUANTIZE_LINEAR, &weightDequantizeOperatorDesc};
-
-        DML_GEMM_OPERATOR_DESC gemmOperatorDesc = {};
-        gemmOperatorDesc.ATensor = inputDequantizeOperatorDesc.OutputTensor;
-        gemmOperatorDesc.BTensor = weightDequantizeOperatorDesc.OutputTensor;
-
-        if (hasBias)
-        {
-            gemmOperatorDesc.CTensor = &inputDescs[2];
-        }
+        DML_MATRIX_MULTIPLY_INTEGER_TO_FLOAT_OPERATOR_DESC matrixMultiplyIntergerToFloatOperatorDesc = {};
+        matrixMultiplyIntergerToFloatOperatorDesc.ATensor = &inputDescs[InputIndex::inputIndex];
+        matrixMultiplyIntergerToFloatOperatorDesc.AScaleTensor = &inputDescs[InputIndex::inputScaleIndex];
+        matrixMultiplyIntergerToFloatOperatorDesc.AZeroPointTensor = &inputDescs[InputIndex::inputZeroPointIndex];
+        matrixMultiplyIntergerToFloatOperatorDesc.BTensor = &inputDescs[InputIndex::weightsIndex];
+        matrixMultiplyIntergerToFloatOperatorDesc.BScaleTensor = &inputDescs[InputIndex::weightScaleIndex];
+        matrixMultiplyIntergerToFloatOperatorDesc.BZeroPointTensor = &inputDescs[InputIndex::weightZeroPointIndex];
+        matrixMultiplyIntergerToFloatOperatorDesc.BiasTensor = hasBias? &inputDescs[InputIndex::biasIndex] : nullptr;
+        matrixMultiplyIntergerToFloatOperatorDesc.OutputTensor = &namedMatMulIntToFloatOutputTensorDesc;
 
-        gemmOperatorDesc.OutputTensor = &namedFirstGemmOutputTensorDesc;
-        gemmOperatorDesc.TransA = DML_MATRIX_TRANSFORM_NONE;
-        gemmOperatorDesc.TransB = DML_MATRIX_TRANSFORM_NONE;
-        gemmOperatorDesc.Alpha = 1.0f;
-        gemmOperatorDesc.Beta = 1.0f;
-        gemmOperatorDesc.FusedActivation = nullptr;
-        const DML_OPERATOR_DESC gemmDesc {DML_OPERATOR_GEMM, &gemmOperatorDesc};
+        const DML_OPERATOR_DESC matrixMultiplyIntergerToFloatDesc {(DML_OPERATOR_TYPE)DML_OPERATOR_MATRIX_MULTIPLY_INTEGER_TO_FLOAT, &matrixMultiplyIntergerToFloatOperatorDesc};
```

Comment:
Prefer `static_cast` here. c-style casts are generally not recommended because they are hard to notice and will transparently do `reinterpret_cast` in the background.

Reply:
Also it's a bit weird/inconsistent to mix `x = {}` and `x {}` here, and most lines use `=`. e.g.:
`DML_INTERMEDIATE_GRAPH_EDGE_DESC mMItoFloatToValueSliceEdge = {};`

Propose:
`const DML_OPERATOR_DESC matrixMultiplyIntergerToFloatDesc = {static_cast<DML_OPERATOR_TYPE>(DML_OPERATOR_MATRIX_MULTIPLY_INTEGER_TO_FLOAT), &matrixMultiplyIntergerToFloatOperatorDesc};`